server/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareServiceImpl.java

```diff
@@ -0,0 +1,687 @@
+
+        Set<State> validStates = new HashSet<>(List.of(State.Ready, State.Stopped));
+        if (!validStates.contains(fileShare.getState())) {
+            throw new InvalidParameterValueException("Stop file share can be done only if the file share is in " + validStates.toString() + " states");
+        }
+
+        if (cleanup == false) {
+            if (!fileShare.getState().equals(State.Stopped)) {
+                stopFileShare(fileShare.getId(), false);
+            }
+            return startFileShare(fileShare.getId());
+        } else {
+            return reDeployFileShare(fileShare, true);
+        }
+    }
+
+    private Pair<List<Long>, Integer> searchForFileSharesIdsAndCount(ListFileSharesCmd cmd) {
+        Account caller = CallContext.current().getCallingAccount();
+        List<Long> permittedAccounts = new ArrayList<>();
+
+        Long id = cmd.getId();
+        String name = cmd.getName();
+        Long networkId = cmd.getNetworkId();
+        Long diskOfferingId = cmd.getDiskOfferingId();
+        Long serviceOfferingId = cmd.getServiceOfferingId();
+        String keyword = cmd.getKeyword();
+        Long startIndex = cmd.getStartIndex();
+        Long pageSize = cmd.getPageSizeVal();
+        Long zoneId = cmd.getZoneId();
+        String accountName = cmd.getAccountName();
+        Long domainId = cmd.getDomainId();
+        Long projectId = cmd.getProjectId();
+
+        Ternary<Long, Boolean, Project.ListProjectResourcesCriteria> domainIdRecursiveListProject = new Ternary<>(domainId, cmd.isRecursive(), null);
+        accountMgr.buildACLSearchParameters(caller, id, accountName, projectId, permittedAccounts, domainIdRecursiveListProject, cmd.listAll(), false);
+        domainId = domainIdRecursiveListProject.first();
+        Boolean isRecursive = domainIdRecursiveListProject.second();
+        Project.ListProjectResourcesCriteria listProjectResourcesCriteria = domainIdRecursiveListProject.third();
+        Filter searchFilter = new Filter(FileShareVO.class, "created", false, startIndex, pageSize);
+
+        SearchBuilder<FileShareVO> fileShareSearchBuilder = fileShareDao.createSearchBuilder();
+        fileShareSearchBuilder.select(null, SearchCriteria.Func.DISTINCT, fileShareSearchBuilder.entity().getId()); // select distinct
+        accountMgr.buildACLSearchBuilder(fileShareSearchBuilder, domainId, isRecursive, permittedAccounts, listProjectResourcesCriteria);
+
+        fileShareSearchBuilder.and("id", fileShareSearchBuilder.entity().getId(), SearchCriteria.Op.EQ);
+        fileShareSearchBuilder.and("name", fileShareSearchBuilder.entity().getName(), SearchCriteria.Op.EQ);
+        fileShareSearchBuilder.and("dataCenterId", fileShareSearchBuilder.entity().getDataCenterId(), SearchCriteria.Op.EQ);
+
+        if (keyword != null) {
+            fileShareSearchBuilder.and().op("keywordName", fileShareSearchBuilder.entity().getName(), SearchCriteria.Op.LIKE);
```

Comment:
should this be `fileShareSearchBuilder.and().op` or `fileShareSearchBuilder.and(` ?

Reply:
.op will open a new parenthesis

Reply:
removed .op and tested